### ISSUE TITLE
apiMetrics not working (nest v8)

### ISSUE THREAD

**Issue by danshapir** [pragmaticivan/nestjs-otel#102]
All the apiMetrics aren't showing up in the metrics page, only the default system ones are.

Using the basic snippet.
```
OpenTelemetryModule.forRoot({
      metrics: {
        hostMetrics: true, // Includes Host Metrics
        defaultMetrics: true, // Includes Default Metrics
        apiMetrics: {
          enable: true, // Includes api metrics
          // timeBuckets: [], // You can change the default time buckets
        },
        defaultLabels: {
          app: process.env.npm_package_name,
          version: process.env.npm_package_version,
        },
      },
    }),
```

if I only have apiMetrics enabled, I get:
```
# no registered metrics
```
as the metrics response. I have no idea how, but one in around 500 tries I saw for a second the needed metrics, but can't reproduce - so in general, it doesn't work.

**Comment by pragmaticivan** [comment]
@danshapir are you using express or fastify ?

**Comment by pragmaticivan** [comment]
Could you share more info about your your project? In the example folder I actually have added an example with nests 8.x.

**Comment by danshapir** [comment]
Using express, pino.
I debugged the code, the API metrics middleware runs and the counters are changing, but not displayed on the metrics endpoint

**Comment by danshapir** [comment]
@pragmaticivan latest version on all packages

**Comment by pragmaticivan** [comment]
@danshapir I'm not able to replicate with any of the examples, would you be able to create a replicable repo I could take a look? Right now it's really hard to figure out what's the problem.

**Comment by danshapir** [comment]
@pragmaticivan My bad! I didnt await the otelSDK.start() in the main.ts.

I'll hijack this issue for another issue. the defaultLabels parameter for the opentelemetryModule doesn't work - it just ignores what I pass in.

**Comment by pragmaticivan** [comment]
Released in 2.5.0.  `defaultLabels` is available inside `apiMetrics` property. Some metric plugins don't allow custom labels yet.
